Bottom: 56409c2ad660f0717dc54fefed8e388c01fa8a01
Top:    fd757ff2ef7cf8485c3e558d91dc74afaaec9617
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-02-13 18:25:20 +0000

Refresh of add-polygon-parentage-tree

---

diff --git a/src/polygon.c b/src/polygon.c
index 8a36e84..be4fcf2 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -136,7 +136,7 @@ polygon_init (void)
 
   /* DEBUG - AVOID PCB running the system out of memory! */
   getrlimit (RLIMIT_AS, &limit);
-  limit.rlim_cur = MIN (limit.rlim_cur, 2000 * 1024 * 1024 /* 2000 GiB limit to virtual memory size */);
+  limit.rlim_cur = MIN (limit.rlim_cur, 7000 * 1024 * 1024 /* 2000 GiB limit to virtual memory size */);
   setrlimit (RLIMIT_AS, &limit);
 
 }
@@ -2361,6 +2361,12 @@ delete_piece_cb (gpointer data, gpointer userdata)
   piece->f->b = piece->b;
   piece->f = piece->b = piece;
 
+  /* Detach the parentage information, so we don't free it.. copies still belong to the M_POLYAREA we are taking this piece from */
+  piece->parentage.immaculate_conception = true;
+  piece->parentage.action = PBO_NONE;
+  piece->parentage.a = NULL;
+  piece->parentage.b = NULL;
+
   poly_Free (&piece);
 }
